From 242e4058f8e7ee6580e00797152389369d4368c9 Mon Sep 17 00:00:00 2001
From: Scott Ellis <scott@jumpnowtek.com>
Date: Sun, 1 Aug 2021 10:21:46 -0400
Subject: [PATCH] Fix sd card boot

---
 include/configs/zynq-common.h | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

diff --git a/include/configs/zynq-common.h b/include/configs/zynq-common.h
index 39035f8beb..c35a04c31b 100644
--- a/include/configs/zynq-common.h
+++ b/include/configs/zynq-common.h
@@ -196,16 +196,22 @@
 /* Default environment */
 #ifndef CONFIG_EXTRA_ENV_SETTINGS
 #define CONFIG_EXTRA_ENV_SETTINGS	\
-	"scriptaddr=0x20000\0"	\
-	"script_size_f=0x40000\0"	\
-	"fdt_addr_r=0x1f00000\0"        \
-	"pxefile_addr_r=0x2000000\0"    \
-	"kernel_addr_r=0x2000000\0"     \
-	"scriptaddr=0x3000000\0"        \
-	"ramdisk_addr_r=0x3100000\0"    \
+	"machine_name=zc706-zynq7\0" \
+	"kernel_image=uImage\0" \
+	"kernel_load_address=0x2080000\0" \
+	"devicetree_image=zynq-zc706.dtb\0" \
+	"devicetree_load_address=0x2000000\0" \
+	"bootargs=earlyprintk console=ttyPS0,115200 root=/dev/mmcblk0p2 rw rootwait\0" \
+	"loadkernel=load mmc 0:2 ${kernel_load_address} /boot/${kernel_image}\0" \
+	"loaddtb=load mmc 0:2 ${devicetree_load_address} /boot/${devicetree_image}\0" \
+	"bootkernel=mmc dev 0 && run loadkernel && run loaddtb && bootm ${kernel_load_address} - ${devicetree_load_address}\0" \
+	"uenvcmd=run bootkernel\0" \
 	BOOTENV
 #endif
 
+#undef CONFIG_BOOTCOMMAND
+#define CONFIG_BOOTCOMMAND "run uenvcmd"
+
 /* Miscellaneous configurable options */
 
 #define CONFIG_CLOCKS
-- 
2.25.1

